src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -92,37 +115,56 @@ object ResourceMatcher {
       badConstraints.isEmpty
     }
 
-    for {
-      cpuRole <- cpuMatchOpt.flatMap(_.matchingRole)
-      memRole <- memMatchOpt.flatMap(_.matchingRole)
-      diskRole <- diskMatchOpt.flatMap(_.matchingRole)
-      portRanges <- portsOpt
-      if meetsAllConstraints
-    } yield ResourceMatch(cpuRole, memRole, diskRole, portRanges)
+    if (scalarMatchResults.forall(_.matches)) {
+      for {
+        portsMatch <- portsMatchOpt
+        if meetsAllConstraints
+      } yield ResourceMatch(scalarMatchResults.collect { case m: ScalarMatch => m }, portsMatch)
+    }
+    else {
+      None
+    }
+  }
+
+  private[this] def matchScalarResource(
```

Comment:
Can you add a comment here or rename the `note` param to something more explanatory? It's not really obvious here what the param is meant for since it's passed along to the `ScalarMatch`